src/sql/workbench/services/queryEditor/browser/queryEditorService.ts

```diff
@@ -99,8 +100,11 @@ export class QueryEditorService implements IQueryEditorService {
 	}
 
 	////// Private functions
+	private createUntitledSqlFilePath(providerName?: string): Promise<string> {
+		if (providerName === 'KUSTO') {
```

Comment:
Shouldn't this just always use the provider name if it's given? Why make Kusto special here?

Reply:
There are other providers like "PGSQL" which we name as sql editors, its done this way to make sure its a safe change. Secondly, later it will easier to track this special casing when we find a better generic solution.